Realm/RLMAccessor.mm

```diff
@@ -1100,7 +1127,7 @@ static auto toOptional(__unsafe_unretained id const value) {
 
     try {
         realm::Object::create(*this, _realm->_realm, *_info.objectSchema,
-                              (id)value, policy, existingKey, outObj);
+                              realm::util::any_cast<id>(value), policy, existingKey, outObj);
```

Comment:
We should not be using std::any anywhere. This is constructing a temporary std::any from value and then reading the value right back out of it.